src/redux/optimismExplorer.js

```diff
@@ -83,30 +82,28 @@ export const optimismExplorerInit = () => async (dispatch, getState) => {
       return;
     }
 
-    const tokenAddresses = assets.map(
-      ({ asset: { asset_code } }) => asset_code
+    const tokenAddresses = assets.map(({ asset: { asset_code } }) =>
+      toLower(asset_code)
     );
 
     const balances = await fetchAssetBalances(
-      assets.map(({ asset: { asset_code } }) => asset_code),
+      tokenAddresses,
       accountAddress,
       network
     );
 
-    let total = BigNumber.from(0);
-
+    let updatedAssets = assets;
     if (balances) {
-      Object.keys(balances).forEach(key => {
```

Comment:
`balances` is already a map between token addresses and balances.
We were going through each of the keys of balances then going through the assets list (O(m*n)).
Now, we just do a map through the assets list (O(n)) and do a lookup in the `balances` map